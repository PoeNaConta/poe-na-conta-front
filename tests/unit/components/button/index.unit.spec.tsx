import React from 'react';
import Button from '@components/button';
import { fireEvent, render, screen } from '@testing-library/react';

describe('Button', () => {
  it.each([{ primary: true }, { secondary: true }])(
    'should render button of type secondary',
    (
      typesProps:
        | { primary: true; secondary?: never }
        | { primary?: never; secondary: true },
    ) => {
      const { container } = render(
        <Button {...typesProps} onClick={vi.fn()}>
          example button
        </Button>,
      );

      expect(container).toMatchSnapshot();
    },
  );

  it('should not render button of type secondary', () => {
    const { container } = render(
      <Button secondary={false} fake>
        example button
      </Button>,
    );

    expect(container).toMatchSnapshot();
    expect(container.firstElementChild).not.toHaveClass(
      'lds--button--secondary',
    );
  });

  it('should render fake button', () => {
    const { container } = render(
      <Button primary fake>
        example button
      </Button>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render button of type %s', () => {
    const onClickMock = vi.fn();
    render(
      <Button primary onClick={onClickMock}>
        example button
      </Button>,
    );

    const button = screen.getByRole('button', { name: 'example button' });
    fireEvent.click(button);

    expect(onClickMock).toBeCalled();
  });
});
