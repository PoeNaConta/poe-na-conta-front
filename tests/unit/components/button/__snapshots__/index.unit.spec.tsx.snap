// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should not render button of type secondary 1`] = `
<div>
  <button
    class="lds--button undefined"
  >
    example button
  </button>
</div>
`;

exports[`Button > should render button of type secondary 1`] = `
<div>
  <button
    class="lds--button lds--button--primary"
  >
    example button
  </button>
</div>
`;

exports[`Button > should render button of type secondary 2`] = `
<div>
  <button
    class="lds--button lds--button--secondary"
  >
    example button
  </button>
</div>
`;

exports[`Button > should render fake button 1`] = `
<div>
  <button
    class="lds--button lds--button--primary"
  >
    example button
  </button>
</div>
`;
